src/ui/screens/system.rs

```diff
@@ -75,6 +75,8 @@ impl ActivatableScreen for SystemScreen {
     fn activate(&mut self, ui: &Ui, display: Display) -> Box<dyn ActiveScreen> {
         draw_border("System Status", SCREEN_TYPE, &display);
 
+        display.with_lock(|target| draw_button_legend(target, "Next / Do", "Screen"));
+
```

Comment:
`draw_border()` followed by `with_lock(|t| draw_button_legend)` seems to be a common pattern now. Check if we should make `draw_border` accept a `DisplayExclusive` like `draw_button_legend` does, so both can be drawn in the same `.with_lock()` call.